### PR TITLE
Added a beat-sensitive show (called Thump) to play with.

### PR DESCRIPTION
**Pull request by xian**

### CODE REVIEW

File: src/commonMain/kotlin/baaahs/shows/ThumpShow.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    override fun nextFrame() {
+        val theta = ((getTimeMillis() / 1000f) % (2 * PI)).toFloat()
+        val beat = beatProvider.beat
+
+        var i = 0
+        shaderBufs.forEach { shaderBuffer ->
+            shaderBuffer.solidShaderBuffer.color = Color.BLACK.fade(colorPicker.color, beat % 1f)
+            shaderBuffer.sineWaveShaderBuffer.color = if (beat < .2) Color.WHITE else Color.ORANGE
```
Comment (aphexcx) [inline src/commonMain/kotlin/baaahs/shows/ThumpShow.kt]: Guessing beat would be .1, .2, .3, .4 and then wraps back to .1 every bar?

File: src/commonMain/kotlin/baaahs/Pinky.kt [hunk 2]
```diff
@@ -96,22 +98,28 @@ class Pinky(
         brainsChanged = true
     }
 
-    inner class BeatProvider(val bpm: Float) {
-        var startTimeMillis = 0L
-        var beat = 0
-        var beatsPerMeasure = 4
+    inner class BeatProvider(var bpm: Float) {
```
Comment (aphexcx) [inline src/commonMain/kotlin/baaahs/Pinky.kt]: Assume the bpm will change every frame. How do we update this var?

I believe the mixer does a continuous sliding window sample and gives its best guess at any point. This is actually even better than relying on a single track-wide value provided by rekordbox or any other track analyzer - those tend to be wrong for the breakdowns or vocal sections of tracks, or tracks that speed up/ slow down a lot. The mixer will actually peg things more accurately in those cases.

Comment (xian) [reply src/commonMain/kotlin/baaahs/Pinky.kt]: Yep, agreed that we should expect it to be continuously changing... whatever the source of truth for beat sync is, it'll need to update those vars somehow; ... how that happens depends on how it's implemented? You update a car by assigning a new value to it. ;-)

File: src/commonMain/kotlin/baaahs/shows/ThumpShow.kt (hunk 1 above)
Comment (xian) [reply src/commonMain/kotlin/baaahs/shows/ThumpShow.kt]: I was thinking 0.0, 1.0, 2.0, 3.0 (for 4/4 time), but it's a float so you get intermediate values too.

**Pull request merged**